src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_StringConcat.cs

```diff
@@ -494,17 +786,39 @@ BoundExpression makeConditionalAccess(BoundExpression receiver)
                     forceCopyOfNullableValueType: false,
                     type: _compilation.GetSpecialType(SpecialType.System_String));
             }
+        }
 
-            static bool isFieldOfMarshalByRef(BoundExpression expr, CSharpCompilation compilation)
+        private static MethodSymbol? FindSpecificToStringOfStructType(TypeSymbol exprType, MethodSymbol objectToStringMethod)
+        {
+            MethodSymbol? structToStringMethod = null;
+            if (exprType.IsValueType && !exprType.IsTypeParameter())
             {
-                Debug.Assert(!IsCapturedPrimaryConstructorParameter(expr));
-
-                if (expr is BoundFieldAccess fieldAccess)
+                var type = (NamedTypeSymbol)exprType;
+                var typeToStringMembers = type.GetMembers(objectToStringMethod.Name);
+                foreach (var member in typeToStringMembers)
                 {
-                    return DiagnosticsPass.IsNonAgileFieldAccess(fieldAccess, compilation);
+                    if (member is MethodSymbol toStringMethod &&
+                        toStringMethod.GetLeastOverriddenMethod(type) == (object)objectToStringMethod)
+                    {
+                        structToStringMethod = toStringMethod;
```

Comment:
>structToStringMethod = toStringMethod; [](http://example.com/codeflow?start=24&length=38)

It looks like the local isn't necessary, we can return right from here, and return null at the end of the method.